### PR TITLE
Adding Article class

### PR DESCRIPTION
**Pull request by eseiver**
This PR forms the basis of redoing many allofplos functions based on a class. It will especially help for the functions that currently utilize the `transformations.py` functions; using an object makes articles DOI-first in their definition and any other representation (url, local XML file, etc) can be easily obtained without 'losing' the original DOI. No functions in `plos_corpus` or `samples.corpus_analysis` are updated to use the class (though I did throw in a few unrelated small updates).

It includes:
- new files defining the Article class (`article_class.py`) with helper functions (`article_elements.py`)
- updated unittests to include Article class tests
- updated allofplos_basics.ipynb to use the class

unrelated to Article class:
- new file `citation_utilities.py`, courtesy of @freelanceastro (converted from py2 from [here](https://github.com/PLOS/rich_citations/blob/master/python-code/citation_utilities.py)), to parse citations using BeautifulSoup and calling the CrossRef API
- add new function `get_local_dois()` in `samples.corpus_analysis`
- fix file path in `plos_corpus`

All unit tests are passing.

### CODE REVIEW

**Comment by eseiver** [comment]
@sbassi I'm showing the Travis checks had failed; is there a `travis.yml` file I need to add to my branch? Happy to do so.

File: allofplos/transformations.py [hunk 1]
```diff
@@ -56,7 +56,7 @@ def filename_to_doi(filename):
     :param directory: defaults to corpusdir, containing article files
     :return: full unique identifier for a PLOS article
     """
-    #import pdb; pdb.set_trace()
+    # import pdb; pdb.set_trace()
```
Comment (sbassi) [inline allofplos/transformations.py]: remove this

**Comment by eseiver** [comment]
OK @sbassi, changes made!

**Comment by eseiver** [comment]
@sbassi any other changes? otherwise, should be good to merge.

**Pull request merged**